source/industry-solutions/rag-applications.txt

```diff
@@ -0,0 +1,348 @@
+<https://www.mongodb.com/products/platform/atlas-vector-search>`__, both native
+capabilities in MongoDB Atlas, this solution walks developers through
+continuously updating, storing, and searching embeddings with a single 
+interface.
+
+While this solution demonstrates creating vector embeddings of song lyrics in
+different languages, the scenario is relevant to many industries and use cases,
+including:
+
+- **Financial services:** Financial documents, legal policies, and contracts
+  often use multiple languages and differ based on country regulations.
+  Empowering loan officers with an AI-powered interface for expediting loan
+  creation can optimize banking workflows; however, the optimization will only
+  benefit as much as the data is relevant and fresh.
+
+- **Healthcare and Insurance:** From constantly updating patient records to 
+  AI-powered underwriting of insurance policies, it’s important that any RAG 
+  system that optimizes these processes has access to the latest information.
+
+- **Retail:** Personalizing retail experiences by delivering the right offer at
+  the right time to the right customer is critical. However, consider the many
+  languages that shoppers might use and product descriptions that have to
+  match. Routing up-to-date, contextual data to the most accurate embedding
+  model can improve these experiences.
+
+Reference architectures
+-----------------------
+
+With MongoDB
+~~~~~~~~~~~~
+
+- **MongoDB:** With a MongoDB cluster deployed in Atlas, it allows you to store
+  the lyrics and related information under the same document (tags, vectors,
+  etc.). Additionally, Atlas provides a vector index to support semantic 
+  searches using the `MongoDB Aggregation Framework <https://www.mongodb.com/basics/aggregation>`__.
+- **Atlas Stream Processing:** This Stream Processing Instance subscribes to
+  the events generated by MongoDB, filters the relevant information, transforms
+  the events, and emits them to the corresponding Kafka topic. Additionally, it
+  will subscribe to the Kafka cluster to update the documents that change.
+- **Confluent Kafka cluster:** This managed Kafka cluster will receive the new
+  documents and updates from current documents to be processed. Additionally,
+  the events received by the processor will be directed to Atlas Stream 
+  Processing.
+
+- **Metadata service:**
+
+  - **Embedding generator:** Python script that subscribes to the Kafka input
+    topics (both Spanish and English). For each message received, it reads the
+    lyrics and generates an embedding using a model specific for each 
+    language.
+  - **Tags extractor:** Python script that extracts the tags from the lyrics
+    (the 10 most common nouns) and adds it to the resulting event.
+
+.. figure:: /includes/images/industry-solutions/RefArchitecture_StreamProcessing_WithMongoDB.svg
+   :figwidth: 1200px
+   :alt: Scalable vector updates reference architecture with MongoDB
+
+Data model approach
```

Comment:
I know the section titles weren't title case in the original but I feel like we should title case here. Just looks weird to me in sentence case 🤣

Reply:
I agree, I was just taking the purist approach on copy edits. I'll update.